File: app/src/main/java/com/udacity/maluleque/meutako/SettingsActivity.java
package com.udacity.maluleque.meutako;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.EditTextPreference;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import java.util.Locale;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.settings, new SettingsFragment())
                .commit();
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    public static class SettingsFragment extends PreferenceFragmentCompat {
        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey);

            FirebaseUser firebaseUser = FirebaseAuth.getInstance().getCurrentUser();

            Preference preference = findPreference(getString(R.string.share_key));
            preference.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
                @Override
                public boolean onPreferenceClick(Preference preference) {
                    Intent sendIntent = new Intent();
                    sendIntent.setAction(Intent.ACTION_SEND);
                    sendIntent.putExtra(Intent.EXTRA_TEXT, "Hey " + getString(R.string.app_name) + " is an amazing app to track your money. Try it now!");
                    sendIntent.setType("text/plain");

                    Intent shareIntent = Intent.createChooser(sendIntent, null);
                    startActivity(shareIntent);
                    return true;
                }
            });
            EditTextPreference user = findPreference(getString(R.string.username_key));
            user.setText(firebaseUser.getDisplayName());

            EditTextPreference currency = findPreference(getString(R.string.currency_key));
            currency.setText(Locale.getDefault().getCountry());
        }
    }
}

File: app/src/main/java/com/udacity/maluleque/meutako/SettingsActivity.kt
package com.udacity.maluleque.meutako

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.EditTextPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.google.firebase.auth.FirebaseUser
import java.util.*
import javax.inject.Inject

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.settings, SettingsFragment())
                .commit()
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
    }

    class SettingsFragment : PreferenceFragmentCompat() {

        @Inject
        lateinit var firebaseUser: FirebaseUser

        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.root_preferences, rootKey)
            val preference = findPreference<Preference>(getString(R.string.share_key))

            (activity?.application as App).appComponent.inject(this)

            preference!!.onPreferenceClickListener = Preference.OnPreferenceClickListener {
                val sendIntent = Intent()
                sendIntent.action = Intent.ACTION_SEND
                sendIntent.putExtra(Intent.EXTRA_TEXT, "Hey " + getString(R.string.app_name) + " is an amazing app to track your money. Try it now!")
                sendIntent.type = "text/plain"
                val shareIntent = Intent.createChooser(sendIntent, null)
                startActivity(shareIntent)
                true
            }
            val user = findPreference<EditTextPreference>(getString(R.string.username_key))
            user!!.text = firebaseUser.displayName
            val currency = findPreference<EditTextPreference>(getString(R.string.currency_key))
            currency!!.text = Locale.getDefault().country
        }
    }
}